services/snapshotter/service_test.go

```diff
@@ -379,6 +381,42 @@ func TestSnapshotter_RequestRetentionPolicyInfo(t *testing.T) {
 	}
 }
 
+func TestSnapshotter_RequestUpdateMeta(t *testing.T) {
+	s, l, err := NewTestService()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer l.Close()
+
+	var tsdbStore internal.TSDBStoreMock
+	tsdbStore.CreateShardFn = func(database, policy string, shardID uint64, enabled bool) error {
+		return nil
+	}
+
+	s.MetaClient = &MetaClient{data: &meta.Data{}}
+	s.TSDBStore = &tsdbStore
+	if err := s.Open(); err != nil {
+		t.Fatalf("unexpected open error: %s", err)
+	}
+	defer s.Close()
+
+	metaBytes, _ := data.MarshalBinary()
+
+	req := &snapshotter.Request{
+		Type:                   snapshotter.RequestMetaStoreUpdate,
+		BackupDatabase:         "db0",
+		RestoreDatabase:        "db1",
+		BackupRetentionPolicy:  "autogen",
+		RestoreRetentionPolicy: "autogen",
+		UploadSize:             int64(len(metaBytes)),
+	}
+
+	c := snapshotter.NewClient(l.Addr().String())
+	if _, err := c.UpdateMeta(req, bytes.NewReader(metaBytes)); err != nil {
```

Comment:
Is there anything useful we could assert about the success case here?

Reply:
Of course, I've add some codes to cover the success case